docs/content/any/guides/data_filtering/index.md

```diff
@@ -0,0 +1,194 @@
+
+`{{% exampleGet "authorizedResources" %}}` returns a list of all the
+resources a user is allowed to do an action on. The results of a built and
+executed query.
+
+`{{% exampleGet "authorizedQuery" %}}` returns the query object itself.
+This lets you add additional filters or sorts or any other data to it before
+executing it.
+
+You must define how to build queries and a few other details when you register classes to enable these methods.
+
+# Implementing data filtering
+
+## Query Functions
+
+There are three Query functions that must be implemented. These define what a query is for your application, how the logic in the policy maps to them, how to execute them and how to combine two queries.
+
+### Build a Query
+
+`{{% exampleGet "buildQuery" %}}` takes a list of `Filter`s and returns a
+`Query`
+
+`Filter`s are individual pieces of logic that must apply to the data being
+fetched.
+
+Filters have a `kind`, a `field` and a `value`. Their meaning depends on the
+`kind` field.
+
+- `Eq` means that the field must be equal to the value.
+- `Neq` means that the field must not be equal to the value.
+- `In` means that the field must be equal to one of the values in value.
+Value will be a list.
+- `Contains` means that the field must contain the value. This only applies
+if the field is a list.
+
+### Execute a Query
+
+`{{% exampleGet "execQuery" %}}` takes a query and returns a list of the results.
+
+### Combine Queries
+
+`{{% exampleGet "combineQuery" %}}` takes two queries and returns a new
+query that returns the union of the other two. For example if the two
+queries are SQL queries combine could `UNION` them. If they were HTTP
+requests `{{% exampleGet "combineQuery" %}}` could put them in an array and 
+`` could handle executing an array of queries
+and combining the results.
+
+You can define functions that apply to all types with
+`{{% exampleGet "setDataFilteringQueryDefaults" %}}`. Or you can pass type specific ones when you register a class.
+
+## Types
+
+The other thing you have to define to use data filtering are types for
+registered classes. These let oso know what the type of the fields are. Oso
+needs this information to handle specializers and other things in the policy
+when we don't have a concrete resource. The types are a 
+{{% exampleGet "map" %}} from field name to type.
+
+# Example
+{{< literalInclude
+      dynPath="exampleAPath"
+      from="docs: begin-a1"
+      to="docs: end-a1"
+      fallback="no" >}}
+
+For each class we need to register it and define the query functions.
+
+{{< literalInclude
+      dynPath="exampleAPath"
+      from="docs: begin-a2"
+      to="docs: end-a2"
+      fallback="no" >}}
+
+Then we can load a policy and query it.
+
+{{< literalInclude
+      dynPath="policyAPath"
+      fallback="no" >}}
+
+{{< literalInclude
+      dynPath="exampleAPath"
+      from="docs: begin-a3"
+      to="docs: end-a3"
+      fallback="no" >}}
+
+# Relations
```

Comment:
If possible it would be great to only include the query builder functions once between these two examples just to shorten the page down a bit.

Reply:
Making this a h1 header causes it and everything after not to show up in the table of contents sidebar. Maybe just make this h2.